backend/app/services/quiz_check.py

```diff
@@ -0,0 +1,30 @@
+from sqlalchemy.ext.asyncio import AsyncSession
+from app.schemas.quiz_check import QuizCheckIn, QuizCheckOut
+from app.services.quiz import get_quiz_by_id
+from collections import defaultdict
+
+
+async def check_files(session: AsyncSession, file_check_in: QuizCheckIn) -> QuizCheckOut:
+    quiz_id = file_check_in.quiz_id
+    db_quiz = await get_quiz_by_id(session, quiz_id)
+    db_smells = db_quiz.code_smells
+    score = len(db_smells)
+
+    incorrect_smells = defaultdict(list)
+    not_found_smells = defaultdict(list)
+
+    response_files = [file for file in file_check_in.files]
+    for response_file in response_files:
+        for smell in response_file.smells:
+            if smell not in db_smells:
```

Comment:
I wonder whether comparing pydantic objects with sqlalchemy objects works as it is expected to work here. Have you tested that on a real test scenario? I would presume in operator might be wrong there but correct me if I'm wrong